### PR TITLE
JCRVLT-684: test coverage and fix for bug in mixin restoration

### PR DESCRIPTION
**Pull request by reschke**

### CODE REVIEW

**Comment by reschke** [comment]
Might need some more documentation, but here we go.

**Comment by reschke** [comment]
@kwin - running the test in isolation works over here, but when running full tests it fails; is that a concurrency issue? (me confused)

**Comment by reschke** [comment]
So it appears the tests fails with Jackrabbit; for now I just made it Oak-only. Please review.

**Comment by reschke** [comment]
The test fails with Jackrabbit because the move of the child node (upon stash restore) fails early; moxing the restoration of properties up seems to fix this.

I would propose to treat this as separate TODO.

File: vault-core/src/test/resources/test-packages/stashing.zip/META-INF/vault/config.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+<!--
```
Comment (kwin) [inline vault-core/src/test/resources/test-packages/stashing.zip/META-INF/vault/config.xml:1]: no need to add that

File: vault-core/src/test/resources/test-packages/stashing.zip/META-INF/vault/definition/.content.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kwin) [inline vault-core/src/test/resources/test-packages/stashing.zip/META-INF/vault/definition/.content.xml:1]: no need to add that

File: vault-core/src/test/resources/test-packages/stashing2.zip/META-INF/vault/config.xml [hunk 3]
```diff
@@ -0,0 +1,86 @@
+<!--
```
Comment (kwin) [inline vault-core/src/test/resources/test-packages/stashing2.zip/META-INF/vault/config.xml:1]: not necessary, please remove

File: vault-core/src/test/resources/test-packages/stashing2.zip/META-INF/vault/definition/.content.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kwin) [inline vault-core/src/test/resources/test-packages/stashing2.zip/META-INF/vault/definition/.content.xml:1]: please remove

**Review by kwin — CHANGES_REQUESTED** [review]
LGTM, overall, just remove the redundant files from the exploded testing packages

**Comment by reschke** [comment]
Will do.

Other ideas:

- move both test packages under common rolder
- change test NS URI to something we could use for all tests (under "https://jackrabbit.apache.org/filevault/"?)

**Review by kwin — APPROVED** [review]

**Pull request merged**